### PR TITLE
user-registration-form

### PR DESCRIPTION
**Pull request by Aumarh**
user registration form using next.js.

### CODE REVIEW

File: frontend/src/pages/signup.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+export default function Signup(props: Props) {
+  const [email, setEmail] = useState('');
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const [error, setError] = useState<Error>();
+  const router = useRouter();
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    console.log('this is the here');
```
Comment (hynnot) [inline frontend/src/pages/signup.tsx:17]: I think we can remove this

File: frontend/src/pages/signup.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    const response = await fetch('/api/user', {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ email, username, password }),
+    });
+    if (response.ok) {
```
Comment (hynnot) [inline frontend/src/pages/signup.tsx:23]: Maybe, it's interesting to show a custom message depending the response. Maybe we could test this.

**Review by hynnot — APPROVED** [review]
I have added a couple of comments that if you want we can do them now or later. What you prefer.

File: frontend/src/pages/signup.tsx (hunk 1 above)
Comment (Aumarh) [reply frontend/src/pages/signup.tsx:17]: Sure

File: frontend/src/pages/signup.tsx (hunk 2 above)
Comment (Aumarh) [reply frontend/src/pages/signup.tsx:23]: Added custom message for password

**Pull request merged**